src/WingetCreateCLI/Commands/UpdateCommand.cs

```diff
@@ -194,14 +194,29 @@ public async Task<Manifests> DeserializeExistingManifestsAndUpdate(List<string>
                 return null;
             }
 
-            if (!PackageParser.UpdateInstallerNodes(installerManifest, this.InstallerUrls, packageFiles, out bool installerMismatch, out WingetCreateCore.Models.Installer.Installer installersMissingMatch))
+            if (!PackageParser.UpdateInstallerNodesAsync(
+                installerManifest,
+                this.InstallerUrls,
+                packageFiles,
+                out bool installerMismatch,
+                out List<WingetCreateCore.Models.Installer.Installer> unmatchedInstallers,
+                out List<WingetCreateCore.Models.Installer.Installer> multipleMatchedInstallers))
             {
                 if (installerMismatch)
                 {
                     Logger.ErrorLocalized(nameof(Resources.MultipleInstallerUpdateDiscrepancy_Error));
-                    if (installersMissingMatch != null)
+                    Logger.ErrorLocalized(nameof(Resources.NewInstallerUrlMustMatchExisting_Message));
+
+                    if (unmatchedInstallers.Any())
+                    {
+                        Logger.ErrorLocalized(nameof(Resources.InstallerMatchFailedError_Message));
+                        unmatchedInstallers.ForEach(i => Logger.WarnLocalized(nameof(Resources.InstallerDetectedFromUrl_Message), i.Architecture, i.InstallerType, i.InstallerUrl));
+                    }
+
+                    if (multipleMatchedInstallers.Any())
                     {
-                        Logger.ErrorLocalized(nameof(Resources.MissingPackageError_Message), installersMissingMatch.InstallerType, installersMissingMatch.Architecture);
+                        Logger.ErrorLocalized(nameof(Resources.MultipleMatchingInstallerNodes_Error));
+                        multipleMatchedInstallers.ForEach(i => Logger.WarnLocalized(nameof(Resources.InstallerDetectedFromUrl_Message), i.Architecture, i.InstallerType, i.InstallerUrl));

```

Comment:
I tried VSCode and found this message to be a bit misleading.

```
Each new installer URL must have a match to an existing installer node based on installer type and architecture.
Multiple existing installer nodes matched the following installer(s):
X86 Inno installer detected from the url: https://az764295.vo.msecnd.net/insider/87d692b7bf1eb5a663b3e62620a6e7e436e1838c/VSCodeUserSetup-ia32-1.59.0-insider.exe
X86 Inno installer detected from the url: https://az764295.vo.msecnd.net/insider/87d692b7bf1eb5a663b3e62620a6e7e436e1838c/VSCodeSetup-ia32-1.59.0-insider.exe
X86 Inno installer detected from the url: https://az764295.vo.msecnd.net/insider/87d692b7bf1eb5a663b3e62620a6e7e436e1838c/VSCodeUserSetup-x64-1.59.0-insider.exe
X86 Inno installer detected from the url: https://az764295.vo.msecnd.net/insider/87d692b7bf1eb5a663b3e62620a6e7e436e1838c/VSCodeSetup-x64-1.59.0-insider.exe
X86 Inno installer detected from the url: https://az764295.vo.msecnd.net/insider/87d692b7bf1eb5a663b3e62620a6e7e436e1838c/VSCodeUserSetup-arm64-1.59.0-insider.exe
X86 Inno installer detected from the url: https://az764295.vo.msecnd.net/insider/87d692b7bf1eb5a663b3e62620a6e7e436e1838c/VSCodeSetup-arm64-1.59.0-insider.exe
```

This makes it seem like maybe the problem is that we detected x86 for all the installers, when in fact with our new URL parsing logic, we actually detect the correct architectures, so showing this makes it seem like a different issue. Perhaps we need to update the Installer node with the arch detected from the URL prior to bailing out of the UpdateInstallerNodesAsync method?

Reply:
I agree that this can cause some confusions. I added some extra logic and messages so that it is clear to the user what the differences are between architecture detections.